var/spack/repos/builtin/packages/xeus/package.py

```diff
@@ -13,31 +13,53 @@ class Xeus(CMakePackage):
     url      = "https://github.com/QuantStack/xeus/archive/0.14.1.tar.gz"
     git      = "https://github.com/QuantStack/xeus.git"
 
+    maintainers = ['tomstitt']
+
     version('develop', branch='master')
+    version('1.0.4', sha256='7324ff013eb97d579fd3b6f9770a13f8863d6046c8bbcdbe2fc7d2ac02f0161f')
     version('0.15.0', sha256='bc99235b24d5757dc129f3ed531501fb0d0667913927ed39ee24281952649183')
     version('0.14.1', sha256='a6815845d4522ec279f142d3b4e92ef52cd80847b512146a65f256a77e058cfe')
 
     variant('examples', default=False, description="Build examples")
+    variant('shared', default=True, description="Build shared libraries")
+    variant('tests', default=False, description="Build tests")
 
     conflicts('%gcc@:4.8')
     conflicts('%clang@:3.6')
     conflicts('%intel@:17')
 
     depends_on('libzmq@4.2.5:-libsodium')
-    depends_on('cppzmq@4.3.0:')
-    depends_on('cryptopp@7.0.0:')
-    depends_on('xtl@0.4.0:')
-    depends_on('nlohmann-json@3.2.0', when='@develop,0.15.0:')
+    depends_on('cppzmq@4.7.1:', when='@1.0.4:,develop')
+    depends_on('cppzmq@4.3.0:', when='@:0.15.0')
+
+    depends_on('cryptopp@7.0.0:', when='@:0.15.0')
+
+    depends_on('openssl@1.0.1:', when='@1.0.4:,develop')
+
+    depends_on('xtl@0.4.0:', when='@:0.15.0')
+    depends_on('xtl@0.7.0:0.7.999', when='@1.0.4:,develop')
+
+    depends_on('nlohmann-json@3.4.0:', when='@1.0.4:,develop')
+    depends_on('nlohmann-json@3.2.0', when='@0.15.0')
     depends_on('nlohmann-json@3.1.1', when='@0.14.1')
-    depends_on('uuid')
+
+    depends_on('uuid', when='platform=linux')
```

Comment:
Is this not needed on macOS? We added a `apple-libuuid` package and a `uuid` virtual dependency specifically to handle this kind of things.

Reply:
I did this based on their README: https://github.com/jupyter-xeus/xeus

Specifically, under [Building from Source](https://github.com/jupyter-xeus/xeus#building-from-source) they list their dependencies then say:
> On Linux platforms, xeus also requires `libuuid`, which is available in all linux distributions (`uuid-dev` on Debian).